core/src/labels/labelCollider.cpp

```diff
@@ -20,35 +20,41 @@ void LabelCollider::addLabels(std::vector<std::unique_ptr<Label>>& _labels) {
     }
 }
 
-void LabelCollider::handleRepeatGroup(size_t startPos) {
+size_t LabelCollider::filterRepeatGroups(size_t startPos, size_t curPos) {
 
-    float threshold2 = pow(m_labels[startPos].label->options().repeatDistance, 2);
-    size_t repeatGroup = m_labels[startPos].label->options().repeatGroup;
+    size_t endGroup = m_labels[curPos].label->options().repeatGroup;
+    size_t endPos = curPos;
 
-    // Get the range
-    size_t endPos = startPos;
-    for (;startPos > 0; startPos--) {
-        if (m_labels[startPos-1].label->options().repeatGroup != repeatGroup) { break; }
-    }
-    for (;endPos < m_labels.size()-1; endPos++) {
-        if (m_labels[endPos+1].label->options().repeatGroup != repeatGroup) { break; }
-    }
+    for (size_t pos = startPos; pos <= curPos; pos = endPos) {
 
+        size_t repeatGroup = m_labels[pos].label->options().repeatGroup;
+        float  threshold2 = pow(m_labels[pos].label->options().repeatDistance, 2);
+
+        // Find end of the current repeatGroup
+        size_t endPos = pos;
```

Comment:
oh right - this should 'only' have caused some labels being checked twice though.